akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolFlow.scala

```diff
@@ -77,14 +77,20 @@ private object PoolFlow {
       import GraphDSL.Implicits._
 
       val conductor = b.add(PoolConductor(maxConnections, pipeliningLimit, log))
-      val slots = Vector
-        .tabulate(maxConnections)(PoolSlot(_, connectionFlow, settings))
-        .map(b.add)
+      //The difference between hot and cold slots:
+      // - hot: should be kept alive, the number is based on "minConnections" from settings
+      // - cold: hold ordinary connection that might be killed, the number is "maxConnections" - "minConnections"
+      val hotSlots = Vector
+        .tabulate(minConnections)(PoolSlot(_, connectionFlow, settings, isHotConnection = true))
+      val coldSlots = Vector
+        .tabulate(maxConnections - minConnections)(PoolSlot(_, connectionFlow, settings, isHotConnection = false))
+      val allSlots = (hotSlots ++ coldSlots).map(b.add)
```

Comment:
I think the notion of "hot" is misleading here. We just want to have "x established connections at all times". I [sketched out](https://github.com/akka/akka/issues/20456#issuecomment-222963595) in the ticket how this can be achieved. It should be the task of the conductor to manage this. The slots shouldn't need to change much.